core/arch/arm/kernel/kern.ld.S

```diff
@@ -211,6 +211,41 @@ SECTIONS
 	. = ALIGN(SMALL_PAGE_SIZE);
 #endif
 	__flatmap_unpg_ro_size = . - __flatmap_unpg_ro_start;
+
+#ifdef CFG_VIRTUALIZATION
+	__flatmap_kern_rw_start = . ;
+	.kdata : ALIGN(8) {
+		*(.kdata .kdata.*)
+	}
+	.kbss : ALIGN(8) {
+		__kbss_start = .;
+		*(.kbss .kbss.*)
+		__kbss_end = .;
+	}
+	.kstack (NOLOAD) : {
+		__nozi_stack_start = .;
+		KEEP(*(.nozi_stack.stack_tmp .nozi_stack.stack_abt))
+		. = ALIGN(8);
+		__nozi_stack_end = .;
+	}
+
+	.kheap (NOLOAD) : {
+		__kheap_start = .;
+		. += CFG_CORE_KHEAP_SIZE;
+		. = ALIGN(16 * 1024);
+		__kheap_end = .;
+	}
+	.knozi (NOLOAD) : {
```

Comment:
I added the following comment:

```	
        /*
	 * We want to keep all kernel memory in one place, because
	 * it should be always mapped and it is easier to map one
	 * memory region than two.
	 * Next section are NOLOAD ones, but they are followed
	 * by sections with data. Thus, this NOLOAD section will
	 * be included in the resulting binary, filled with zeroes
	 */

```